### ISSUE TITLE
FP8 precision padding error

### ISSUE THREAD

**Issue by yunbaohe2022** [SinanAkkoyun/lit-llama-fp8#1]
Did you use the padding in the commented code when doing FP8 precision text generation inference? Would it cause errors if used?

Your Comments
"""
    # SINAN: FP8 token multiple of 8:
    # Ensure the encoded sequence's length is a multiple of 8
    remainder = len(encoded) % 8
    if remainder != 0:
        padding_needed = 8 - remainder
        padding_tensor = torch.full((padding_needed,), tokenizer.pad_id, dtype=torch.long, device=fabric.device)
        encoded = torch.cat((encoded, padding_tensor))
    """

**Comment by yunbaohe2022** [comment]
I look forward to your response.

**Comment by SinanAkkoyun** [comment]
Hi @yunbaohe2022 !
I stopped working on this as the original repo reworked it and have their own (working, mine is not) implementation

**Comment by yunbaohe2022** [comment]
Thanks for your reply.
